suites/apis/dataUploadTest.js

```diff
@@ -266,7 +267,75 @@ Scenario('Upload the same file twice @dataUpload', async (sheepdog, indexd, node
   );
 }).retry(2);
 
-BeforeSuite(async (dataClient, fence, users, sheepdog, indexd, files) => {
+/**
+ * Use fence's multipart upload endpoints to upload a large data file (>5MB)
+ */
+Scenario('Multipart upload', async (files, users, fence, indexd, dataUpload) => {
+  // create file and file parts
+  // each part must be >=5MB (except the last part)
+  const bigFileName = '7mb_file.txt';
+  await files.createBigTmpFile(bigFileName, 7);
+  const fileContents = fs.readFileSync(bigFileName);
+
+  const fiveMbLength = Math.floor(fileContents.length * 5/7);
+  const fileParts = {
+    1: fileContents.slice(0, fiveMbLength),
+    2: fileContents.slice(fiveMbLength)
+  };
+
+  // initialize the multipart upload
+  console.log("Initializing multipart upload");
+  const accessHeader = users.mainAcct.accessTokenHeader;
+  const initRes = await fence.complete.initMultipartUpload(bigFileName, accessHeader);
+  const fileGuid = initRes.guid;
+  createdGuids.push(fileGuid);
+  const key = `${fileGuid}/${bigFileName}`;
+
+  // upload each file part to the S3 bucket
+  let partsSummary = [];
+  for (var partNumber = 1; partNumber <= Object.keys(fileParts).length; partNumber++) {
+    console.log(`Uploading file part ${partNumber}`);
+
+    // get a presigned URL from fence
+    let getUrlRes = await fence.complete.getUrlForMultipartUpload(key, initRes.uploadId, partNumber, accessHeader);
+
+    // upload the file part using the presigned URL
+    const uploadPartRes = await dataUpload.uploadFilePartToS3(
+      getUrlRes.url,
+      fileParts[partNumber]
+    );
+
+    partsSummary.push({
+      PartNumber: partNumber,
+      ETag: uploadPartRes
+    });
+  }
+
+  // complete the multipart upload
+  console.log("Completing multipart upload");
+  await fence.complete.completeMultipartUpload(key, initRes.uploadId, partsSummary, accessHeader);
+
+  // wait for the indexd listener to add size, hashes and URL to the record
+  let fileNode = {
+    did: fileGuid,
+    data: {
+      md5sum: await files.getFileHash(bigFileName),
+      file_size: files.getFileSize(bigFileName)
+    }
+  };
+  await dataUpload.waitUploadFileUpdatedFromIndexdListener(indexd, fileNode);
+
+  // download the file
+  console.log("Downloading the file");
+  const signedUrlRes = await fence.do.createSignedUrlForUser(fileGuid);
+  fence.ask.assertStatusCode(signedUrlRes, 200, "Could not get signed URL for file download");
+  fence.complete.checkFileEquals(signedUrlRes, fileContents.toString());
```

Comment:
I prefer to check hash and size than than check the entire object content since it follows our logic but it is OK for me if you want to keep that line

Reply:
the hash and size are checked as well, during `waitUploadFileUpdatedFromIndexdListener`